src/main/java/com/microsoft/sqlserver/jdbc/ConfigurableRetryLogic.java

```diff
@@ -283,16 +282,25 @@ private static void createConnectionRules(LinkedList<String> listOfRules) throws
     private static String getCurrentClassPath() throws SQLServerException {
         String location = "";
         String className = "";
+        String locationSuffix = "target/classes/";
 
         try {
             className = new Object() {}.getClass().getEnclosingClass().getName();
             location = Class.forName(className).getProtectionDomain().getCodeSource().getLocation().getPath();

```

Comment:
If I do a `java -jar mssql-jdbc.jar` with the above code, the location always includes the JAR name eg. `/path/to/mssql-jdbc.jar`. But, when I run it in my IDE I get `/path/to/`. Without the JAR name in the path.

Do you need to account for the cases mentioned?

Reply:
`Class.forName(className).getProtectionDomain().getCodeSource().getLocation().getPath();` will return the `/path/to` example you showed above, i.e. where the jar is without including the jar. This is the scenario I tested before pushing this PR.